bsp/espressif/esp/src/hals/esp32-c3/gpio.zig

```diff
@@ -0,0 +1,148 @@
+const std = @import("std");
+const microzig = @import("microzig");
+const peripherals = microzig.chip.peripherals;
+const IO_MUX = peripherals.IO_MUX;
+const GPIO = peripherals.GPIO;
+
+pub const Level = enum(u1) {
+    low = 0,
+    high = 1,
+};
+
+pub const DriveStrength = enum(u2) {
+    @"5mA" = 0,
+    @"10mA" = 1,
+    @"20mA" = 2,
+    @"40mA" = 3,
+};
+
+pub const Pin = struct {
+    number: u5,
```

Comment:
Since the only field in this struct is a `u5`, you can actually just turn this into an `enum(u5)` instead of a struct! Then, any call to `self.number` can become `@intFromEnum(self)`.

Reply:
True, but then that would create bigger changes down the line if we want to add things to the struct. I will definitely think about this though!

Reply:
For sure, personal preference!